scan/src/pages/DataSourceHomePage.re

```diff
@@ -106,57 +122,32 @@ let make = () => {
           <Col> <HSpacing size=Spacing.xl /> </Col>
         </Row>
       </THead>
-      {[
-         (
-           1,
-           "CoinGecko V.2",
-           MomentRe.momentNow(),
-           "band17rprjgtj0krfw3wyl9creueej6ca9dc4dgxv6e" |> Address.fromBech32,
-           123.3,
-         ),
-         (
-           2,
-           "Koo",
-           MomentRe.momentNow()
-           |> MomentRe.Moment.subtract(~duration=MomentRe.duration(2., `hours)),
-           "band17rprjgtj0krfw3wyl9creueej6ca9dc4dgxv6e" |> Address.fromBech32,
-           123.3,
-         ),
-         (
-           3,
-           "Binance",
-           MomentRe.momentNow()
-           |> MomentRe.Moment.subtract(~duration=MomentRe.duration(10., `hours)),
-           "band17rprjgtj0krfw3wyl9creueej6ca9dc4dgxv6e" |> Address.fromBech32,
-           123.3,
-         ),
-         (
-           4,
-           "CMC",
-           MomentRe.momentNow()
-           |> MomentRe.Moment.subtract(~duration=MomentRe.duration(22., `hours)),
-           "band17rprjgtj0krfw3wyl9creueej6ca9dc4dgxv6e" |> Address.fromBech32,
-           123.3,
-         ),
-       ]
-       ->Belt.List.map(((id, name, timestamp, owner, fee)) => {
-           <TBody key=name>
-             <div className=Styles.fullWidth>
-               <Row>
-                 <Col> <HSpacing size=Spacing.xl /> </Col>
-                 <Col size=0.5>
-                   <TElement elementType={TElement.DataSource(ID.DataSource.ID(id), name)} />
-                 </Col>
-                 <Col size=0.5> <TElement elementType={timestamp->TElement.Timestamp} /> </Col>
-                 <Col size=1.> <TElement elementType={owner->TElement.Address} /> </Col>
-                 <Col size=0.4> <TElement elementType={fee->TElement.Fee} /> </Col>
-                 <Col> <HSpacing size=Spacing.xl /> </Col>
-               </Row>
-             </div>
-           </TBody>
-         })
-       ->Array.of_list
-       ->React.array}
+      {switch (dataSourcesOpt) {
+       | Some(dataSource) =>
+         dataSource
+         ->Belt.List.map(({id, name, timestamp, owner, fee}) => {
+             <TBody key=name>
+               <div className=Styles.fullWidth>
+                 <Row>
+                   <Col> <HSpacing size=Spacing.xl /> </Col>
+                   <Col size=0.5>
+                     <TElement elementType={TElement.DataSource(ID.DataSource.ID(id), name)} />
+                   </Col>
+                   <Col size=0.5> <TElement elementType={timestamp->TElement.Timestamp} /> </Col>
+                   <Col size=1.> <TElement elementType={owner->TElement.Address} /> </Col>
+                   <Col size=0.4>
+                     <TElement elementType={fee->TxHook.Coin.getFeeAmount->TElement.Fee} />
```

Comment:
Expect to reuse Fee type? or float type?

Reply:
![image](https://user-images.githubusercontent.com/26691764/76597671-97a2fc80-6533-11ea-95c3-4713f20dd50f.png)